doc/man1/flux-exec.rst

```diff
@@ -67,6 +67,16 @@ OPTIONS
 
    Exclude ranks in *NODESET*.
 
+.. option:: -j, --jobid=JOBID
+
+   Use the job shell exec service for job *JOBID* instead of the broker's
+   exec service. The default *NODESET* will be set to the nodes assigned
+   to the job, and the :option:`--rank` and :option:`--exclude` options are
+   applied relative to this set. For example, :option:`flux exec -j ID -r 0`
+   will run only on the first node assigned to *JOBID*, and :option:`flux
+   exec -j ID -x 0` will run on all nodes assigned to *JOBID* except the
+   first node.
```

Comment:
It feels like this paragraph needs a stronger statement that the ranks are shell ranks not broker ranks.

Reply:
We changed the use of "job ranks" here based on your previous comment:
(My assumption is the casual reader will not know what "job shell ranks" means, so I instead gave examples)

> In the docs, "job ranks" might be misunderstood as corresponding to a task rather than a shell. Maybe "job node ranks" or "job shell ranks" or similar would be better?

> Actually I see that "job ranks" and {ranks} is used to refer to shell ranks in [flux-jobs(1)](https://flux-framework.readthedocs.io/projects/flux-core/en/latest/man1/flux-jobs.html#output-format) also. Hmm, maybe for now we should just be consistent.

Feel free to rewrite however you think is best.

Reply:
BTW, I forgot to mention that the way `flux exec` works was changed slightly by some of the commits I added here. Previously, the "nodeset" (as it is called in he man page) is initialized via the `-r, --ranks=` option or it set to "all" ranks using the size attribute.

Now the nodeset is always initialized to the set of all broker ranks, unless `-j, --jobid` is used in which case it is initialized to the set of ranks assigned to the job. Then the `-r, --ranks` and `-x, --exclude` options are used to include/exclude ranks within that set by index. This works the same as before for `-r, --ranks` (since the index matches the broker rank for this case) _except_ that you can no longer specify invalid ranks (which seemed an improvement, but maybe there's an argument otherwise?) It also allows `-r` and `-x` to be used to filter the target ranks with `--jobid` as well since otherwise you'd have to know a priori the ranks assigned to the job, which wouldn't be usable IMO.

Reply:
Another thought - the `flux-exec(1)` manpage is the only one that seems to use the term "nodeset". Probably this was one of the early manpages. Maybe that term doesn't make sense anymore, and this manpage needs a complete rewrite with modern terminology? We could point to a definition of "job shell rank" in the documentation of `--jobid`, but AFAICS that term doesn't appear anywhere else in the flux-core docs, so maybe the whole manual could be cleaned up with consistent terminology?